File: DataStructuresAndLibraries/pagina01/1162/1162.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(){
	
	int cases;
	cin >> cases;

	for(int i = 0; i < cases; i++){

		int len, count = 0;
		vector<int> vet;

		cin >> len;
		for(int n= 0; n < len; n++){ //capturando dados do vetor
			int value;
			cin >> value;
			vet.push_back(value);
		}

		//implementando o algoritmo de bubble sort.
		for(int j =0; j < len - 1; j++){
			for(int k=0; k < len - j - 1; k++){
				if(vet[k] > vet[k + 1]){
					count++;
					int temp = vet[k];
					vet[k] = vet[k + 1];
					vet[k + 1] = temp;
				}
			}
		}

		cout << "Optimal train swapping takes " << count << " swaps." << endl;
		
	}
	
	return 0;
}

File: DataStructuresAndLibraries/__1000__/1162/1162.java
import java.io.IOException;
import java.util.Scanner;

public class Main{
    public static void main(String[] args) throws IOException {

        Scanner input = new Scanner(system.in);

        int n = input.nextInt();
        input.nextLine();

        for(int i = 0; i < n; i++){
            int len = input.nextInt();
            int count = 0;

            int[] vet = new int[len];
            for(int j= 0; j < len; j++){
                vet[j] = input.nextInt();
            }

            for(int l = 0; l < len - 1; l++){
                for(int k = 0; k < len - l - 1; k++){
                    if(vet[k] > vet[k + 1]){
                        count++;
                        int temp = vet[k];
                        vet[k] = vet[k + 1];
                        vet[K + 1] = temp;
                    }
                }
            }

            System.out.println("Optimal train swapping takes " + count + " swaps.");
        }

        input.close();
    }
}